osquery/devtools/shell.cpp

```diff
@@ -125,32 +140,84 @@ static sqlite3_int64 timeOfDay(void) {
 }
 
 // Saved resource information for the beginning of an operation
+#ifdef WIN32
+static ULARGE_INTEGER sKernelTimeBegin;
+static ULARGE_INTEGER sUserTimeBegin;
+#else
 static struct rusage sBegin; // CPU time at start
+#endif
+
 static sqlite3_int64 iBegin; // Wall-clock time at start
 
 static void beginTimer(void) {
   if (enableTimer) {
+#ifdef WIN32
+    FILETIME ftCreation = {0};
+    FILETIME ftExit = {0};
+    FILETIME ftKernel = {0};
+    FILETIME ftUser = {0};
+
+    ::GetProcessTimes(::GetCurrentProcess(), &ftCreation, &ftExit, &ftKernel,
+                      &ftUser);
+
+    sKernelTimeBegin.HighPart = ftKernel.dwHighDateTime;
+    sKernelTimeBegin.LowPart = ftKernel.dwLowDateTime;
+
+    sUserTimeBegin.HighPart = ftUser.dwHighDateTime;
+    sUserTimeBegin.LowPart = ftUser.dwLowDateTime;
+#else
     getrusage(RUSAGE_SELF, &sBegin);
+#endif
+
     iBegin = timeOfDay();
   }
 }
 
 // Return the difference of two time_structs in seconds
+#ifdef WIN32
+static double timeDiff(ULARGE_INTEGER *pStart, ULARGE_INTEGER *pEnd) {
```

Comment:
Should we consider making an issue to build out abstraction classes for small utilitarian type functions such as these? It seems that small things like this keep showing up, and I wonder if it'd be nice for us to have all of these small helper functions centralized and abstracted?